scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -211,6 +221,44 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
   // NOTE: Special case for Enum region is needed because parsing of 'case' statement is done differently
   case object RegionEnumArtificialMark extends SepRegion
 
+  case class AllowColonEol() {
+    private var unclosed = 0
+    def unclosedParen = unclosed
+    def openParen = unclosed += 1
+    def closeParen = unclosed -= 1
+  }
+
+  /* Heuristic to try to determine if colonEol can exist at a given point in code
+   * needed to deal with situations like:
+   * def a(b:
+   * Int)
+   */
+  def canBeFollowedByColonEol(
+      colonEolCanStartIndent: Option[AllowColonEol],
+      curr: Token
+  ) = {
+    if (dialect.allowSignificantIndentation)
+      colonEolCanStartIndent match {
+        case _ if curr.is[CanStartColonEol] =>
+          Some(AllowColonEol())
+        // This means we closed more parents than opened, which means we are outside of a region where colonEol is allows
```

Comment:
`.. is allowed`

Reply:
I think comment for particular case should be inside this case, maybe subjective, [suggestion]

e.g.:
https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/parsing/Scanners.scala#L194

Reply:
Changed to `is allowed`, but kept it here, I think it makes sense.